### ISSUE TITLE
Getting Same Class ID During Inferencing

### ISSUE THREAD

**Issue by immohan6598** [Ipsedo/MARLClassification#6]
Hi,

Great work!

I have trained a new model with 2 class using the 0 and 1 class images from mnist and got a 90% precision and 90% Recall 
but when i do inference on single image always getting predicted class id 0
![confusion_matrix_epoch_39_eval](https://github.com/Ipsedo/MARLClassification/assets/89785217/15cd9f04-4d42-4e78-898d-90608776263a)
[marl.json](https://github.com/Ipsedo/MARLClassification/files/13968089/marl.json)

![animated_gif](https://github.com/Ipsedo/MARLClassification/assets/89785217/ebb365ca-88d3-4302-90c3-49b39c1f26af)
![animated_gif](https://github.com/Ipsedo/MARLClassification/assets/89785217/df97e025-8fdf-4358-bfbd-2855735e4bed)

predicted class and the confidence is coming same.

i have trained on custom data as well by increasing the number of agents and observations still same issue.

but at the end of the training the result looks different
![animated_gif](https://github.com/Ipsedo/MARLClassification/assets/89785217/5fb30c82-e78b-4e61-9494-e5670e473ffb)

Thanks

**Comment by immohan6598** [comment]
it looks like we are having issue in saving the trained model or loading the model for inference

**Comment by immohan6598** [comment]
i fixed it 
it is due to model.eval() is not called in the inference.py

**Comment by Ipsedo** [comment]
Maybe it is due to batch normalization if you infer on small batch size (eg 1 data per batch). Thank to see it, I always infer on multiple images so I never face to this :p
This afternoon I will push the fix for inference script
